### PR TITLE
Validations unit test

### PR DESCRIPTION
**Pull request by tiagomarin**
### :heavy_check_mark: Add validations for the User model:
- Name must not be blank.
- PostsCounter must be an integer greater than or equal to zero.
### :heavy_check_mark: Add validations for the Post model:
- Title must not be blank.
- Title must not exceed 250 characters.
- CommentsCounter must be an integer greater than or equal to zero.
- LikesCounter must be an integer greater than or equal to zero.
### :heavy_check_mark:  Add unit tests
- Add unit specs for all of your models' methods and validations.

### :speech_balloon: Feel free to shoot me a message in the pull request if you have any questions.

Thank you for your time!

### CODE REVIEW

File: spec/models/post_spec.rb [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+    it 'should have a key "text" = This is my first post' do
+      expect(post.text).to eql('This is my first post')
+    end
+
+    it 'should belong to a user with "name" = Jonh' do
+      expect(post.author.name).to eql('Jonh')
+    end
+
+    it 'should have a key "comments_counter" = 0' do
+      expect(post.comments_counter).to eql(0)
```
Comment (Munsa1) [inline spec/models/post_spec.rb]: - [x] Kindly add a test case to check for the comments counter to be an integer greater than or equal to 0, as this is a requirement. Currently it only checks if it is equal to 0

File: spec/models/post_spec.rb [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+    end
+
+    it 'should have a key "likes_counter" = 0' do
+      expect(post.likes_counter).to eql(0)
+    end
```
Comment (Munsa1) [inline spec/models/post_spec.rb]: - [x] Kindly add a test case to check for the likes counter to be an integer greater than or equal to 0, as this is a requirement. Currently it only checks if it is equal to 0

File: spec/models/user_spec.rb [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+    end
+    it 'should have a key photo = https://photoUrl.com' do
+      expect(user.photo).to match('https://photoUrl.com')
+    end
+    it 'should have a key bio = Teacher from Mexico' do
+      expect(user.bio).to match('Teacher from Mexico')
+    end
+    it 'should have a key posts_counter = 0' do
+      expect(user.posts_counter).to match(0)
+    end
```
Comment (Munsa1) [inline spec/models/user_spec.rb]: - [x] Kindly add a test case to check for the posts counter to be an integer greater than or equal to 0, as this is a requirement. Currently it only checks if it is matches 0

**Review by Munsa1 — CHANGES_REQUESTED** [review]
Hi @tiagomarin, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!
<FEEL FREE TO CHANGE THIS MESSAGE TO YOUR PERSONAL WORDS>

## Highlights
:heavy_check_mark: Descriptive Readme
:heavy_check_mark: Passing Linters
:heavy_check_mark: Correct Gitflow
## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

You can also consider:

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: spec/models/post_spec.rb (hunk 1 above)
Comment (tiagomarin) [reply spec/models/post_spec.rb]: I've done it!

Thanks for pointing that out!

File: spec/models/post_spec.rb (hunk 2 above)
Comment (tiagomarin) [reply spec/models/post_spec.rb]: I've done it!

Thanks for pointing that out!

File: spec/models/user_spec.rb (hunk 3 above)
Comment (tiagomarin) [reply spec/models/user_spec.rb]: I've done it!

Thanks for pointing that out!

**Review by danielufeli — APPROVED** [review]
Hi @tiagomarin ,

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

a

## Highlights
- Nice commits messages✔️
- Descriptive PR ✔️
- Good use of GitHub flow ✔️
- Clean Code✔️

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me @danielufeli in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by EricMbouwe — COMMENTED** [review]
Hi @tiagomarin, 

While you made a great effort in this project, I cannot proceed to review your code.

## Invalid Code Review Request ⛔❌

You have submitted an already merged PR. You may request another review with the correct PR

Your Code Review Request will be marked as invalid in your Dashboard, so please submit a new one once you are ready 🙏

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

--------

_Invalid Code Review Request **does not count** into the code reviews limit._

**Review by bimbolabuari — COMMENTED** [review]
Hello @tiagomarin 👋,

_This is an additional review as a result of the second opinion request._

- It is confirmed that you already completed this milestone and resubmitted for re-approval, I could also see that a code reviewer already approved on the dashboard but did not leave a review on the PR, however, if you are not able to submit the PR on your dashboard you can resubmit for approval

As this was the only reason that your project was not approved it will be approved as soon as possible.
Please ask for another review in your Dashboard and the next Code Reviewer will mark it as approved right away :)

Keep rocking 🚀!

**Pull request merged**